db/db.go

```diff
@@ -2888,6 +2907,33 @@ func GetTotalBLSChanges() (uint64, error) {
 	return count, nil
 }
 
+func GetBLSChangesCountForQuery(query string) (uint64, error) {
+	bquery, _ := hex.DecodeString(strings.TrimPrefix(query, "0x"))
+	count := uint64(0)
+	err := ReaderDb.Get(&count, `
+		SELECT count(*)
```

Comment:
**issue:** Same issue as with the withdrawals.

Reply:
applied the new logic also for counting